### ISSUE TITLE
Calling submit handler with another dispatched action

### ISSUE THREAD

**Issue by cyberwombat** [davidkpiano/react-redux-form#449]
I have this code that returns a thunk for my API - I am working off a codebase that uses this setup extensively and am wondering if I can keep it in this context (mainly for the convenience of having params from `getState` and `axios`). 

    import { SUBSCRIBE_REQUEST, SUBSCRIBE_SUCCESS, SUBSCRIBE_FAILURE } from '../../constants'

    export function subscribeUser (data) {
      return (dispatch, getState, { axios }) => {
        
        dispatch({ type: SUBSCRIBE_REQUEST })
        return axios.get(`${protocol}://${host}/api/v0/newsletter/subscribe`, data)
          .then(res => {
            dispatch({
              type: SUBSCRIBE_SUCCESS,
              payload: res.data
            })
          })
          .catch(error => {
            dispatch({
              type: SUBSCRIBE_FAILURE,
              payload: error,
              error: true
            })
          })
      }
    }

I am now implementing `react-redux-form` and am unclear on how ot connect the above with the submit function. From the docs it expects me to pass a promise in the submit action:

    class myForm extends React.Component {

      handleSubmit (data) {
        const { dispatch } = this.props   
        dispatch(actions.submit('user', somePromise))

      }

Basically something like:

     dispatch(actions.submit('user', subscribeUser))

How can I connect my thunked API code with the submit handler? I'm getting started with the redux ecosystem so still floundering. Perhaps you have a suggestion. 

Thank you for your time.

**Comment by davidkpiano** [comment]
One way to do it would be to pass callbacks to your `subscribeUser` function:

```js
export function subscribeUser(data, callback, errCallback) {
  
  // ...
  .then(res => {
    dispatch(...);
    callback && callback(res);
  })
  .catch(err => {
    dispatch(...);
    errCallback && errCallback(err);
  });
}
```

And then calling it like so:

```js
class myForm extends React.Component {
  handleSubmit(data) {
    const { dispatch } = this.props;

    const promise = new Promise((resolve, reject) => {
      dispatch(subscribeUser(data,
        res => resolve(),
        err => reject(err)
      ));
    });

    dispatch(actions.submit(promise));
  }
}
```

There might be a cleaner way to do this but this is the first thing that I thought of.

**Comment by cyberwombat** [comment]
Ok thank you for you answer. I wanted to make sure I wasn't missing some obvious way.

**Comment by amyiee** [comment]
Hi,
Got a question regarding to this,
Well i keep on getting this error whenever i implemented this code to my code base
`Uncaught (in promise) Error: Actions must be plain objects. Use custom middleware for async actions.`

Just wondering if you know any middleware, etc to remedied the issue.

Thanks
<img width="1015" alt="screen shot 2017-03-02 at 11 51 20 am" src="https://cloud.githubusercontent.com/assets/498342/23488242/74076cfa-ff3f-11e6-96f1-a3af389f11e9.png">

**Comment by davidkpiano** [comment]
@amyiee Yes - [redux-thunk](https://github.com/gaearon/redux-thunk) should do the trick.

**Comment by amyiee** [comment]
@davidkpiano Got it working, Thanks! 👍
